Refactor/gadelshin timur project recreation

urweakness opened the pull request:
# Что сделано?
- Удалено все ненужное из проекта
- Пересмотрена файловая иерархия
- Добавлены ассеты (цвета, иконки, картинки, шрифты, шрифты иконок)
- Реализовано ядро навигации через паттерн Coordinator
- Реализована Observable обертка над сетевым клиентом для реактивного отслеживания состояния загрузки
- Добавлены модели Requests для каждого рабочего эндпоинта
- Добавлены модели Response для каждого рабочего эндпоинта
- Добавлены UIComponents с переиспользуемыми элементами

# Состояние
- [ + ] проект собирается
- [ - ] варнинги (TODO: TOKEN)

# UI
- поддержка всех тем (light dark), текущая тема управляется системой

urweakness committed:
[Refactor] unnecessaries removal

urweakness committed:
[Refactor] ProgressHUD removal + unnecessaries

urweakness committed:
[Refactor] project settings setting up according to tech task + info.plist iOS 26 UI restriction

urweakness committed:
[Refactor] App file moved to App folder

urweakness committed:
[Refactor] DateFormatters
- moved to extensions
- replaced by constant instead of variable

urweakness committed:
[Refactor] NFTStorage
- rename
- concurrency safely Sendable added

urweakness committed:
[Feature] DataLoader
- added to track loading state during any network request

urweakness committed:
[Refactor] NetworkClient
- concurrent sefyle Sendable added
- send method with decoding is Decodable & Sendable now and receives decode type as parameter

urweakness committed:
[Refactor] NetworkRequest
- concurreny safely Sendable added

urweakness committed:
[Refactor] NFT
- rename
- Hashable now

urweakness committed:
[Feature] ObservedNetworkClient
- added
- contains all endpoint methods
- fetches and tracks loadingState

urweakness committed:
[Refactor] RequestConstants + Requests
- moved to Networking folder

urweakness committed:
[Refactor] NFTService
- renamed
- Sendable safety
- moved up to Services folder

urweakness committed:
[Feature] Resources
- assets added (Assets + Images + Fonts)
- Page, Sheet added

urweakness committed:
[Feature] Modules
- NFTDetail added (NFT Preview)
- AboutAuthor WebView added (shared flow)
- UIComponents added (shared elements)
- TabView added (main TabView)
- Coordinator added (main navigation view)

urweakness committed:
[Refactor] xcode project settings update accepted

urweakness committed:
[Refactor] shopping cart tab removal (added content to shopping cart)
- Images updated

urweakness committed:
[Feature] DecompositionStatistics.md added

numpadCrusader reviewed (approved):
Была проделана крутейшая работа!

urweakness committed:
[Refactor] Dialogs
- unified + simplified confirmationDialogs with self state supporting

urweakness committed:
[Refactor] Alerts
- rename

urweakness committed:
[Refactor] ProfilePhotoActionsViewModifier
- titleVisibility change to .visible

urweakness committed:
[Refactor] BaseConfirmationDialogViewModifier
- animation incapsulated
 - activeSortOption is button now, supports confirmationDialog open

urweakness committed:
[Refactor] Base View
- base Profile, Statistics, Catalog added

urweakness committed:
[Refactor] Splash screen added as FullScreen in root
- confirmation dialogs supports toolbar/safeAreaTop placement now

urweakness committed:
[Refactor] BaseConfirmationDialogViewModifier
- little padding added to make it simple position as in toolbar

urweakness committed:
[Feature] NFTModel added with mocks

urweakness committed:
[Feature] NFTCellLayout added
- allows layout ui elements in NFTCells

urweakness committed:
[Refactor] RatingPreview
- shadow theme adapt

urweakness committed:
[Refactor] NFTListView
- removal due to useless

urweakness committed:
[Refactor] NFTCollectionView
- mocks added
- receives callbacks

urweakness committed:
[Refactor] NFTHorizontalCell
- decomposed by simplified views
- removal due to unnecessary

urweakness committed:
[Refactor] NFTVerticalCell
- moved to components

urweakness committed:
[Feature] NFTCell UI components implemented with usage previews

urweakness committed:
[Refactor] DecomositionStatistics.md
- module 2, 3 updated

urweakness committed:
[Refactor] NetwortRequest
- simplified

urweakness committed:
[Refactor] NFT -> NFTResponse rename

urweakness committed:
[Feature] RequestConstants
- token added

urweakness committed:
[Feature] Response models added

urweakness committed:
[Feature] Request models added

urweakness committed:
[Refactor] ObservedNetworkClient
- methods return values changed
- payloads added

urweakness committed:
[Refactor] NetworkClient
- payload content-type changed

urweakness committed:
[Feature] ObservedNetworkClientTests added
- put order and pay fails

urweakness committed:
[Feature] NFTModel
- Hashable

urweakness committed:
[Refactor] OrderRequests
- put order and pay payload change

urweakness committed:
[Refactor] ObservedNetworkClientTests
- method rename
- order refactor

urweakness committed:
[Feature] LaunchScreen added
- supports any appearance

urweakness committed:
[Refactor] Network Layer
- correnct format mapper added
- finalized net core

urweakness committed:
[Refactor] DecompositionStatistics.md
